### PR TITLE
fix #2290 by using the HashCodeCombiner from Common

### LINKED ISSUE #2290: Move HashCodeCombiner to Common repo
We should move the `HashCodeCombiner` class to the Common repo as a Sources package to avoid duplication.

### PR DESCRIPTION
**Pull request by anurse**
Now that HashCodeCombiner has been moved, this changes our code to use the shared sources version.

@victorhurdugaci can this go in before your renames as well?

/cc @troydai @BrennanConroy

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Looks good. Are you doing the same for other repos?

**Comment by anurse** [comment]
@NTaylorMullen said he'd take Razor and that's the only other place this occurs.

**Comment by troydai** [comment]
:shipit:

File: src/Microsoft.Framework.Runtime.Abstractions/project.json [hunk 1]
```diff
@@ -2,6 +2,12 @@
     "version": "1.0.0-*",
     "description": "ASP.NET 5 interfaces to take advantage of capabilities exposed by the runtime.",
     "compilationOptions": { "warningsAsErrors": true },
+    "dependencies": {
+        "Microsoft.Framework.HashCodeCombiner.Sources": {
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: Ugh, lets not invent new formatting. We don't do this anywhere else.

Comment (anurse) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: VS did it to me automatically...

Comment (NTaylorMullen) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: :+1: for having whatever VS chooses here. Less work when having to undo.

Comment (anurse) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: See also: https://github.com/aspnet/Razor/pull/463/files#diff-3ca14b98c54139a0defc8a35940fe07e

Comment (Eilon) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: As ugly as it is, fighting the VS defaults is an uphill battle.

Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: We need to fix this. It's ugly as crap /cc @madskristensen

Comment (anurse) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: Yeah, we shouldn't reformat objects that are already on a single line. Much like C# does with properties:

```csharp
public string Foo { get { ... } set { ... } }
```

Comment (madskristensen) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: @davidfowl What needs to be fixed? VS auto-formats C# or JSON?

Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: JSON

Comment (madskristensen) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: I suggest you take a look at using .editorconfig files to control tab vs. spaces and indent size

Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Abstractions/project.json]: I think you missed the point of this thing. Andrew pretty much summed it up https://github.com/aspnet/dnx/pull/2359#discussion_r35613141

**Pull request closed without merging**